sandbox/rocky/tf/plotter/plotter.py

```diff
@@ -0,0 +1,89 @@
+import atexit
+from multiprocessing import Process
+import numpy as np
+import platform
+from queue import Empty, Queue
+import tensorflow as tf
+from threading import Thread
+
+from rllab.sampler.utils import rollout
+
+
+class Plotter(object):
+    def __init__(self, env, policy):
+        self.env = env
+        self.policy = policy
+        self.sess = tf.get_default_session()
+        self.thread = None
+        self.queue = None
+        self._init_plot()
+
+    def _init_plot(self):
+        self._init_worker()
+        self.queue.put(['update', self.env, self.policy])
+        self.queue.task_done()
+
+    def _init_worker(self):
+        if not self.queue:
+            self.queue = Queue()
+            self.thread = Thread(
+                target=self._start_worker, args=(
+                    self.queue,
+                    self.sess,
+                ))
+            self.thread.daemon = True
+            self.thread.start()
+
+    def _start_worker(self, queue, session):
+        env = None
+        policy = None
+        max_length = None
+        sess = session
+        try:
+            with sess.as_default():
+                with sess.graph.as_default():
+                    while True:
+                        msgs = {}
+                        # Only fetch the last message of each type
+                        while True:
```

Comment:
It seems now GIL is putting both this worker thread and the main thread in the same processor, so this busy waiting will impact performance.
A suggestion on how to avoid busy waiting can be found in my attempt to solve this problem [here](https://github.com/gonzaiva/rllab/blob/0c5bc6731c3da64ee70874a5e12726358d5408ed/sandbox/rocky/tf/plotter/plotter.py#L44-L55).
Feel free to use the code there.